ports/esp32/machine_pwm.c

```diff
@@ -310,31 +462,36 @@ STATIC void mp_machine_pwm_init_helper(machine_pwm_obj_t *self,
     self->channel = CHANNEL_IDX_TO_CHANNEL(channel_idx);
 
     // New PWM assignment
-    if ((chans[channel_idx].pin == -1) || (chans[channel_idx].timer_idx != timer_idx)) {
+    // if ((chans[channel_idx].pin == -1) || (chans[channel_idx].timer_idx != timer_idx)) {
+    // Always reconfigure the channel to remap Pin.OUT mode as PWM output mode
+    // p=PWM(Pin(5, Pin.OUT), 1000, duty_ns=5000)  # works here
+    // p=PWM(Pin(5, Pin.OUT), 1000, duty_ns=5000)  # locks up
+    {
         configure_channel(self);
         chans[channel_idx].pin = self->pin;
     }
```

Comment:
Now the channel is reconfigured always, but I think this can produce artefacts in the PWM wave at this point.